strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/filter/ArtifactSizeFilter.java

```diff
@@ -0,0 +1,56 @@
+package org.carlspring.strongbox.filter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MaxUploadSizeExceededException;
+
+import javax.inject.Inject;
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+@Order(1)
+public class ArtifactSizeFilter implements Filter
+{
+
+    @Inject
+    private MultipartConfigElement multipartConfigElement;
+
+
+    @Override
+    public void init(final FilterConfig filterConfig) throws ServletException
+    {
+        //do nothing
+    }
+
+    @Override
+    public void doFilter(final ServletRequest request,
+                         final ServletResponse response,
+                         final FilterChain chain) throws IOException, ServletException
+    {
+        HttpServletRequest req = (HttpServletRequest) request;
+        String lengthHeader = req.getHeader("Content-Length");
+        if(lengthHeader != null && !isSizeWithinBoundaries(lengthHeader))
```

Comment:
Please, apart from nullability, check that `lengthHeader` is not empty. You can check both not null and not empty with:
`StringUtils.isNotEmpty(lengthHeader)`

Reply:
Done